### ISSUE TITLE
Before hook called multiple times inside one test block

### ISSUE THREAD

**Issue by JohnnyTheBoy** [cypress-io/cypress#17935]
### Current behavior

when writing multiple tests inside one block (describe), before hook triggers multiple times.

### Desired behavior

Before hook inside one block should be triggered only once.

### Test code to reproduce

describe("Testing", () => {

    before(() => {
        cy.log("Before hook is called!");
    });

    beforeEach(() => {
        cy.log("Before each hook is called");
    });

    it("do something", () => { expect(true).to.be.eq(true); });
    it("do something", () => { expect(true).to.be.eq(true); });
    it("do something", () => { expect(true).to.be.eq(true); });
    it("do something", () => { expect(true).to.be.eq(true); });
    it("do something", () => { expect(true).to.be.eq(true); });
    it("do something", () => { expect(true).to.be.eq(true); });
    it("do something", () => { expect(true).to.be.eq(true); });
    it("do something", () => { expect(true).to.be.eq(true); });
    it("do something", () => { expect(true).to.be.eq(true); });
    it("do something", () => { expect(true).to.be.eq(true); });

    it("says hi", () => {
        cy.visit("www.google.com");
        cy.log("Hi!");
    });

    it("do something", () => { expect(true).to.be.eq(true); });
    it("do something", () => { expect(true).to.be.eq(true); });
    it("do something", () => { expect(true).to.be.eq(true); });

});

### Cypress Version

8.3.1

**Comment by JohnnyTheBoy** [comment]
![cypress_before_hook_issue](https://user-images.githubusercontent.com/8955489/131355887-6773195b-4883-4dc6-a175-d224e5577711.png)

**Comment by jennifer-shehane** [comment]
Looks like a duplicate of https://github.com/cypress-io/cypress/issues/17705 where hooks can be rerun in certain situations when there is a top navigation. We are working on this issue currently.
